iroh-net/src/discovery/static_discovery.rs

```diff
@@ -0,0 +1,126 @@
+//! A static discovery implementation that allows adding info for nodes manually.
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    sync::{Arc, RwLock},
+};
+
+use futures_lite::stream::{self, StreamExt};
+use iroh_base::{
+    key::NodeId,
+    node_addr::{AddrInfo, NodeAddr},
+};
+
+use super::{Discovery, DiscoveryItem};
+
+/// A static discovery implementation that allows adding info for nodes manually.
+#[derive(Debug, Default)]
+#[repr(transparent)]
+pub struct StaticDiscovery {
```

Comment:
bikeshedding: at some point we complained that all the discovery services have confusing names and that they should be consistently named something like `ThingPublisher` or `ThingProvider`.  Calling them `ThingDiscovery` is redundant as it already is in the discovery module.

So `StaticProvider`?